homeassistant/components/zha/core/gateway.py

```diff
@@ -126,6 +126,18 @@ def register_entity_reference(
             self._devices[zigpy_device.ieee] = zha_device
         return zha_device
 
+    async def async_device_became_available(
+            self, sender, is_reply, profile, cluster, src_ep, dst_ep, tsn,
+            command_id, args):
+        """Handle tasks when a device becomes available."""
+        await self.async_update_device(sender)
+
+    async def async_update_device(self, sender):
```

Comment:
Will this be used by something else or can we inline it with the method above?

Reply:
I want to keep it separate because the function `async_device_became_available` above will be the place where network level tasks are called.

Reply:
Why does this need to be a coroutine?

Reply:
are you saying to drop async from the def because it’s run in a task?

Reply:
Yes, we don't seem to need to await anything in the function, so then it could just be a normal function.

Reply:
Makes sense. I’ll update it